services/account/apps/worker/src/main.ts

```diff
@@ -13,85 +13,49 @@ BigInt.prototype['toJSON'] = function () {
 
 const logger = new Logger('worker_main');
 
-// bootstrap() does not have a main loop to keep the process alive.
-// We create a promise that never resolves to keep the process alive and processing events.
-const p = new Promise((resolve, reject) => {});
+/*
+ * Shutdown timer will forcibly terminate the app if it doesn't complete
+ * a graceful shutdown when requested. This is mostly because the @nestjs/bullmq
+ * package doesn't seem to behave on exit under certain conditions (mainly if it
+ * can't connect to Redis at startup)
+ */
+function startShutdownTimer() {
+  setTimeout(() => process.exit(1), 10_000);
+}
 
 async function bootstrap() {
-  // createApplicationContext() does not return when the redis connection is not available.
-  // Starting the timeout here to ensure that the application does not hang indefinitely.
-  const createTimeout = (message: string, duration: number): NodeJS.Timeout =>
-    setTimeout(() => {
-      logger.error(message);
-      process.exit(1);
-    }, duration);
-
-  let redisConnectTimeout: NodeJS.Timeout | null = createTimeout('Redis connection timeout!', 30_000);
-
-  const app = await NestFactory.createApplicationContext(WorkerModule, {
-    logger: process.env.DEBUG ? ['error', 'warn', 'log', 'verbose', 'debug'] : ['error', 'warn', 'log'],
-  });
-  logger.log('Nest ApplicationContext created successfully.');
-
   process.on('uncaughtException', (error) => {
     console.error('****** UNCAUGHT EXCEPTION ******', error);
     process.exit(1);
   });
 
+  const app = await NestFactory.createMicroservice(WorkerModule, {
```

Comment:
Just curious, is this new for v10?

Reply:
No, it always existed. Using it was recommended as a solution for how to spin up a NestJS app that doesn't actually listen on any ports or connections, but still runs continuously as a service, while also having it properly initialize the NestJS-based frameworks surrounding schedulers and event emitters.